pkg/skaffold/config/util.go

```diff
@@ -183,14 +183,27 @@ func GetInsecureRegistries(configFile string) ([]string, error) {
 }
 
 func isDefaultLocal(kubeContext string) bool {
-	return kubeContext == constants.DefaultMinikubeContext ||
+	if kubeContext == constants.DefaultMinikubeContext ||
 		kubeContext == constants.DefaultDockerForDesktopContext ||
-		kubeContext == constants.DefaultDockerDesktopContext ||
-		IsKindCluster(kubeContext)
+		kubeContext == constants.DefaultDockerDesktopContext {
+		return true
+	}
+
+	isKind, _ := IsKindCluster(kubeContext)
+	return isKind
 }
 
-func IsKindCluster(kubeContext string) bool {
-	return strings.HasSuffix(kubeContext, "@kind")
+// IsKindCluster checks that the given `kubeContext` is talking to `kind`.
+// It also returns the name of the `kind` cluster.
+func IsKindCluster(kubeContext string) (bool, string) {
+	switch {
```

Comment:
Could you add some doc here about kind naming prefixes with their version?  When nobody is using < 0.6 then can we remove the `kind-` prefixing?